processing/src/main/java/org/apache/druid/query/GroupByMergedQueryRunner.java

```diff
@@ -187,7 +189,7 @@ private void waitForFutureCompletion(
     }
     catch (InterruptedException e) {
       log.warn(e, "Query interrupted, cancelling pending results, query id [%s]", query.getId());
-      future.cancel(true);
+      GuavaUtils.cancelAll(true, futures);
```

Comment:
nit: `future` should be canceled on exceptions too. This is nit since this class is used only by groupBy v1 which is deprecated.

Reply:
Thanks for your tip, i forgot it and now it have been fixed.